test/parser/parseTest.js

```diff
@@ -1,25 +1,78 @@
 import { expect } from 'chai';
+import fs from 'file-system';
 
 import parse from '../../src/parser/parse';
 
 describe('parser/parse()', () => {
+  const wdl = `
+import "tasks.wdl"
+import "sub_workflow.wdl" as SubWorkflow
 
-  it('requires to parse WDL', () => {
-    const src = `
-workflow foo {
+workflow RootWorkflow {
+    File? wfInput
+    File? wfInputTwo
+    File? wfInputThree
+
+    call tasks.TaskOne {
+        input:
+            taskInput = wfInput
+    }
+
+    call SubWorkflow.Workflow {
+        input:
+            wf_input = TaskOne.task_output,
+            wf_input_two = wfInput
+    }
+
+    call SubWorkflow.Workflow as WorkflowAliasOne {
+        input:
+            wf_input = TaskOne.task_output,
+            wf_input_two = wfInputTwo
+    }
+
+    call SubWorkflow.Workflow as WorkflowAliasTwo {
+        input:
+            wf_input = TaskOne.task_output,
+            wf_input_two = wfInputThree
+    }
+
+    output {
+        String output_1 = WorkflowAliasTwo.output_string
+        String? output_2 = Workflow.output_string
+        String? output_3 = WorkflowAliasOne.output_string
+    }
 }`;
 
+  it('requires to parse WDL', () => {
+    const src = 'workflow foo {}';
+
     expect(() => parse(src)).to.not.throw(Error);
-    expect(parse(src).status).to.equal(true);
+    return parse(src).then((res) => {
+      expect(res.status).to.equal(true);
+    });
   });
 
   it('does not support different formats, only WDL', () => {
-    const src = `
-foo bar {
-}`;
+    const src = 'foo bar {}';
 
-    expect(() => parse(src, {
-      format: 'cwl',
-    })).to.throw(Error);
+    return parse(src, { format: 'cwl' }).catch((e) => {
+      expect(e).to.be.an('Error');
+    });
```

Comment:
Unfortunately the logic of the assertion has changed and it tests almost nothing now. If `parse()` completes successfully the test will succeed (but it should fail). Consider using `chai-as-promised` plugin for this and other tests to test promise-based interface in more intuitive manner.

Reply:
Fixed